<!DOCTYPE html>
<html lang="en">
<head profile="http://www.w3.org/2005/10/profile">
    <meta charset="UTF-8">
    <meta name="description" content="">
    <meta name="author" content="">
    <meta name=viewport content="width=device-width, initial-scale=1">
    <title>2.2.4 Loading and comparing data
</title>
    <link rel="stylesheet" href="/css/style.css">
    <link rel="stylesheet" href="/css/syntax.css">
    <link href="https://cdn.jsdelivr.net/npm/daisyui@3.1.0/dist/full.css" rel="stylesheet" type="text/css" />
    <script src="https://cdn.tailwindcss.com"></script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.css" integrity="sha384-GvrOXuhMATgEsSwCs4smul74iXGOixntILdUW9XmUC6+HX0sLNAK3q71HotJqlAn" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.js" integrity="sha384-cpW21h6RZv/phavutF+AuVYrr+dA8xD9zs6FwLpaCct6O9ctzYFfFr4dgmgccOTx" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/contrib/auto-render.min.js" integrity="sha384-+VBxd3r6XgURycqtZ117nYw44OOcIax56Z4dCRWbxyPt0Koah1uHoK0o4+/RRE05" crossorigin="anonymous"
        onload="renderMathInElement(document.body);"></script>

</head>
<body>
    <header>
        <div class="navbar bg-base-100">
          <div class="navbar-start">
            <nav>
            <ol class="inline-flex items-center space-x-1 md:space-x-3">
        <li class="inline-flex items-center">
          <a href="/" class="inline-flex items-center text-l font-medium text-gray-700 hover:text-blue-600 dark:text-gray-400 dark:hover:text-white">
            <svg aria-hidden="true" class="w-4 h-4 mr-2" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"></path></svg>
            Juan Raphael Diaz Sim√µes
          </a>
        </li>
    
        <li>
      <div class="flex items-center">
        <svg aria-hidden="true" class="w-6 h-6 text-gray-400" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" clip-rule="evenodd"></path></svg>
        <a href="/courses" class="ml-1 text-l font-medium text-gray-700 hover:text-blue-600 md:ml-2 dark:text-gray-400 dark:hover:text-white">Courses</a>
      </div>
    </li>

          </div>
          <div class="navbar-end">
            <ul class="menu menu-horizontal px-1">
              <li><a href="/posts">Posts</a></li>
              <li><a href="/courses">Courses</a></li>
              <li><a href="/cv.html">CV</a></li>
              <li><a href="/projects.html">Projects</a></li>
            </ul>
          </div>
        </nav>
    </header>
    <div class="flex justify-center bg-base-100">
<article class="prose max-w-prose mb-5">
  <div class="book-navigation">

    <div class="book-navigation-parent drawer">
      <input id="my-drawer" type="checkbox" class="drawer-toggle" />
      <div class="drawer-content">
        <label for="my-drawer">
          <div class="space-y-1 inline-block">
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
          </div>
          <b>Index</b>
        </label>
      </div>
      <div class="drawer-side">
        <label for="my-drawer" class="drawer-overlay"></label>
           <div class="menu menu-vertical p-4 h-full bg-base-200">
             <h1 class="text-xl mb-5 w-100">
                  Introduction to Bioinformatics

             </h1>
             <div style="flex-basis:100%; height: 0;"}></div>
             <ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/index.html">1 Lectures
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/03-lecture-01/index.html">1.1 Bioinformatics Basics
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/07-lecture-02/index.html">1.2 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/10-lecture-03/index.html">1.3 Comparing Gene Sequences 2
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/13-lecture-04/index.html">1.4 Gene Finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/16-lecture-05/index.html">1.5 Locating genes and its functions: techniques, problems and difficulties
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/index.html">2 Practical Lessons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/index.html">2.1 Basic R and generic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/01-seminar-01/index.html">2.1.1 Basic R
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/02-seminar-02/index.html">2.1.2 Files
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/04-seminar-03/index.html">2.1.3 Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/05-seminar-04/index.html">2.1.4 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/06-control-01/index.html">2.1.5 Loading and processing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/20-seminar-12/index.html">2.1.6 Programming Exercises
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/index.html">2.2 Basic analysis of genetic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/08-seminar-05/index.html">2.2.1 GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/09-seminar-06/index.html">2.2.2 Subseries GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/11-seminar-07/index.html">2.2.3 Creating dot plots
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/14-control-02/index.html">2.2.4 Loading and comparing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/index.html">2.3 Codons and gene finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/15-seminar-09/index.html">2.3.1 Translating bases to codons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/17-seminar-10/index.html">2.3.2 Codon frequencies and finding genes
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/18-control-03/index.html">2.3.3 Theoretical Control
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/19-seminar-11/index.html">2.3.4 Aminoacid sequences and base sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/index.html">2.4 Summary and exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/21-seminar-13/index.html">2.4.1 Theoretical Questions
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/22-exam-01/index.html">2.4.2 Final exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li></ul></li></ul>
           </div>
      </div>
    </div>

    <div class="book-navigation-parent">
      <div class="arrow arrow-up"></div>
      <a href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/index.html">2.2 Basic analysis of genetic data
</a>
      <div class="arrow arrow-up"></div>
    </div>

    <div class="book-navigation-section">

       <div class="book-navigation-previous">
         <div class="arrow arrow-left"></div>
         <a href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/11-seminar-07/index.html">2.2.3 Creating dot plots
</a>
       </div>

       <div class="book-navigation-next">
       </div>
    </div>
  </div>

    <h1 class="text-3xl mb-5">
    2.2.4 Loading and comparing data

</h1>

<p class="mb-3">In this control, you are going to do, by yourselves, repeat the analyses we did in the last classes. After most steps, you will have to execute some code, either by pressing enter in the console, or by clicking the <code>Run</code> button in the text area. If you see any errors (text in red), call me.</p>
<p class="mb-3">You have <strong>30 minutes</strong>. The information you need to do the exam is in the text, <strong>read it attentively</strong>. You can look at the pages of the previous lessons if you want. <strong>Call me</strong> after you succeed each step.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="1" id="using-paper"><span class="header-section-number">1</span> Using paper</h1>
<p class="mb-3">Ask the teacher for two DNA strands. You will create the dot plot such as the one below, but <strong>using your own sequences</strong>.</p>
<p class="mb-3"><img class="mx-auto" src="data/6263efd5f082b523c955ac3348e4da13.png" /></p>
<h1 class="text-2xl mt-5 mb-5" data-number="2" id="loading-the-library"><span class="header-section-number">2</span> Loading the library</h1>
<p class="mb-3">In order to use the <code>seqinr</code> library to load and process gene sequence data, run the following command in the <em>console</em>:</p>
<div class="sourceCode mb-3 notranslate" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a class="link link-hover text-zinc-600" href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span></code></pre></div>
<p class="mb-3"><em>How to know it worked</em>: RStudio will accept your command, and not produce any errors. The line after the command will be just empty.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="3" id="reading-the-data"><span class="header-section-number">3</span> Reading the data</h1>
<p class="mb-3">Save the <code>dengue.fasta</code> data to the <code>dengue</code> variable, and the <code>zika.fasta</code> data to the <code>zika</code> variable. Remembering, that in order to save the <code>lalala.fasta</code> data to the <code>lalala</code> variable, one must do:</p>
<div class="sourceCode mb-3 notranslate" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a class="link link-hover text-zinc-600" href="#cb2-1" aria-hidden="true" tabindex="-1"></a>lalala <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">read.fasta</span>(<span class="st">&quot;lalala&quot;</span>)[[<span class="dv">1</span>]])</span></code></pre></div>
<p class="mb-3"><em>How to know it worked</em>: On the top right panel, in the <code>Environment</code> tab, you will see the variables <code>dengue</code> and <code>zika</code> with the description <code>'SeqFastadna' ...</code>. If you type either <code>dengue</code> or <code>zika</code>, you will see the sequence of nucleotides.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="4" id="calculating-gc-contents"><span class="header-section-number">4</span> Calculating GC contents</h1>
<p class="mb-3">Calculate the GC content of the <code>dengue</code> and <code>zika</code> sequences, and save them to the <code>dengueGC</code> and <code>zikaGC</code> variables. In order to do this, use the <code>lalalaGC &lt;- GC(lalala)</code> pattern.</p>
<p class="mb-3">When both variables are saved, create a plot with:</p>
<div class="sourceCode mb-3 notranslate" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a class="link link-hover text-zinc-600" href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">c</span>(<span class="at">dengue=</span>dengueGC,<span class="at">zika=</span>zikaGC))</span></code></pre></div>
<p class="mb-3"><em>How to know it worked</em>: You should see a bar plot on the bottom-right panel.</p>
<p class="mb-3"><strong>Question</strong>: Which DNA strand is more heat resistant: zika or dengue?</p>
<h1 class="text-2xl mt-5 mb-5" data-number="5" id="creating-a-dot-plot-1"><span class="header-section-number">5</span> Creating a dot plot 1</h1>
<p class="mb-3">Create a dot plot comparing the first 100 nucleotides of the dengue virus to themselves. To select these, you can use <code>dengue[1:100]</code>. The dot plot can be created for a <code>lalala</code> sequence as follows:</p>
<pre><code>dotPlot(lalala,lalala)</code></pre>
<p class="mb-3"><em>How to know it worked</em>: You should see a bar plot on the bottom-right panel.</p>
<p class="mb-3"><strong>Question</strong>: By looking at the image, how do you know this is a dot plot for two equal sequences?</p>
<h1 class="text-2xl mt-5 mb-5" data-number="6" id="creating-a-dot-plot-2"><span class="header-section-number">6</span> Creating a dot plot 2</h1>
<p class="mb-3">Create a dot plot comparing the zika and dengue sequences as follows:</p>
<div class="sourceCode mb-3 notranslate" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a class="link link-hover text-zinc-600" href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotPlot</span>(dengue[<span class="dv">9000</span><span class="sc">:</span><span class="dv">10000</span>], zika[<span class="dv">9000</span><span class="sc">:</span><span class="dv">10000</span>], <span class="at">wsize =</span> <span class="dv">5</span>, <span class="at">nmatch =</span> <span class="dv">5</span>, <span class="at">wstep =</span> <span class="dv">5</span>)</span></code></pre></div>
<p class="mb-3">reminding that the <code>5</code> used in the function call means we will only mark with a dot matches with a size of 5 or more.</p>
<p class="mb-3"><em>How to know it worked</em>: You should see a bar plot on the bottom-right panel.</p>
<p class="mb-3"><strong>Question</strong>: What parts of these subsequences look like each other?</p>
<h1 class="text-2xl mt-5 mb-5" data-number="7" id="creating-a-dot-plot-3"><span class="header-section-number">7</span> Creating a dot plot 3</h1>
<p class="mb-3">Coming back to the DNA strands you compared on paper. Save <strong>your personal DNA sequences</strong> in R:</p>
<div class="sourceCode mb-3 notranslate" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a class="link link-hover text-zinc-600" href="#cb6-1" aria-hidden="true" tabindex="-1"></a>dna1 <span class="ot">&lt;-</span> <span class="fu">s2c</span>(<span class="st">&quot;actcgctgc&quot;</span>)</span>
<span id="cb6-2"><a class="link link-hover text-zinc-600" href="#cb6-2" aria-hidden="true" tabindex="-1"></a>dna2 <span class="ot">&lt;-</span> <span class="fu">s2c</span>(<span class="st">&quot;cgctgcact&quot;</span>)</span></code></pre></div>
<p class="mb-3">and use <code>dotPlot</code> to build the dot plot. Compare with the one you have on paper.</p>

  <ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul>


</article>
</div>

</body>
</html>
