<!DOCTYPE html>
<html lang="en">
<head profile="http://www.w3.org/2005/10/profile">
    <meta charset="UTF-8">
    <meta name="description" content="">
    <meta name="author" content="">
    <meta name=viewport content="width=device-width, initial-scale=1">
    <title>2.2.2 Subseries GC Content
</title>
    <link rel="stylesheet" href="/css/style.css">
    <link rel="stylesheet" href="/css/syntax.css">
    <link href="https://cdn.jsdelivr.net/npm/daisyui@3.1.0/dist/full.css" rel="stylesheet" type="text/css" />
    <script src="https://cdn.tailwindcss.com"></script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.css" integrity="sha384-GvrOXuhMATgEsSwCs4smul74iXGOixntILdUW9XmUC6+HX0sLNAK3q71HotJqlAn" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.js" integrity="sha384-cpW21h6RZv/phavutF+AuVYrr+dA8xD9zs6FwLpaCct6O9ctzYFfFr4dgmgccOTx" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/contrib/auto-render.min.js" integrity="sha384-+VBxd3r6XgURycqtZ117nYw44OOcIax56Z4dCRWbxyPt0Koah1uHoK0o4+/RRE05" crossorigin="anonymous"
        onload="renderMathInElement(document.body);"></script>

</head>
<body>
    <header>
        <div class="navbar bg-base-100">
          <div class="navbar-start">
            <nav>
            <ol class="inline-flex items-center space-x-1 md:space-x-3">
        <li class="inline-flex items-center">
          <a href="/" class="inline-flex items-center text-l font-medium text-gray-700 hover:text-blue-600 dark:text-gray-400 dark:hover:text-white">
            <svg aria-hidden="true" class="w-4 h-4 mr-2" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"></path></svg>
            Juan Raphael Diaz Simões
          </a>
        </li>
    
        <li>
      <div class="flex items-center">
        <svg aria-hidden="true" class="w-6 h-6 text-gray-400" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" clip-rule="evenodd"></path></svg>
        <a href="/courses" class="ml-1 text-l font-medium text-gray-700 hover:text-blue-600 md:ml-2 dark:text-gray-400 dark:hover:text-white">Courses</a>
      </div>
    </li>

          </div>
          <div class="navbar-end">
            <ul class="menu menu-horizontal px-1">
              <li><a href="/posts">Posts</a></li>
              <li><a href="/courses">Courses</a></li>
              <li><a href="/cv.html">CV</a></li>
              <li><a href="/projects.html">Projects</a></li>
            </ul>
          </div>
        </nav>
    </header>
    <div class="flex justify-center bg-base-100">
<article class="prose max-w-prose mb-5">
  <div class="book-navigation">

    <div class="book-navigation-parent drawer">
      <input id="my-drawer" type="checkbox" class="drawer-toggle" />
      <div class="drawer-content">
        <label for="my-drawer">
          <div class="space-y-1 inline-block">
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
          </div>
          <b>Index</b>
        </label>
      </div>
      <div class="drawer-side">
        <label for="my-drawer" class="drawer-overlay"></label>
           <div class="menu menu-vertical p-4 bg-base-200">
             <h1 class="text-xl mb-5 w-100" style="margin-left: 0.75em;">
                  Introduction to Bioinformatics

             </h1>
             <div style="margin-left:-1.5em;">
               <ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/index.html">1 Lectures
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/03-lecture-01/index.html">1.1 Bioinformatics Basics
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/07-lecture-02/index.html">1.2 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/10-lecture-03/index.html">1.3 Comparing Gene Sequences 2
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/13-lecture-04/index.html">1.4 Gene Finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/16-lecture-05/index.html">1.5 Locating genes and its functions: techniques, problems and difficulties
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/index.html">2 Practical Lessons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/index.html">2.1 Basic R and generic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/01-seminar-01/index.html">2.1.1 Basic R
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/02-seminar-02/index.html">2.1.2 Files
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/04-seminar-03/index.html">2.1.3 Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/05-seminar-04/index.html">2.1.4 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/06-control-01/index.html">2.1.5 Control: Loading and processing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/20-seminar-12/index.html">2.1.6 Programming Exercises
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/index.html">2.2 Basic analysis of genetic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/08-seminar-05/index.html">2.2.1 GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/09-seminar-06/index.html">2.2.2 Subseries GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/11-seminar-07/index.html">2.2.3 Creating dot plots
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/14-control-02/index.html">2.2.4 Loading and comparing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/index.html">2.3 Codons and gene finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/15-seminar-09/index.html">2.3.1 Translating bases to codons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/17-seminar-10/index.html">2.3.2 Codon frequencies and finding genes
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/18-control-03/index.html">2.3.3 Theoretical Control
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/19-seminar-11/index.html">2.3.4 Aminoacid sequences and base sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/index.html">2.4 Summary and exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/21-seminar-13/index.html">2.4.1 Theoretical Questions
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/22-exam-01/index.html">2.4.2 Final exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li></ul></li></ul>
             </div>
           </div>
      </div>
    </div>

    <div class="book-navigation-parent">
      <div class="arrow arrow-up"></div>
      <a href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/index.html">2.2 Basic analysis of genetic data
</a>
      <div class="arrow arrow-up"></div>
    </div>

    <div class="book-navigation-section">

       <div class="book-navigation-previous">
         <div class="arrow arrow-left"></div>
         <a href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/08-seminar-05/index.html">2.2.1 GC Content
</a>
       </div>

       <div class="book-navigation-next">
         <a href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/11-seminar-07/index.html">2.2.3 Creating dot plots
</a>
         <div class="arrow arrow-right"></div>
       </div>
    </div>

    <div id="google_translate_element" style="margin-top:1em;"></div>

  </div>

    <h1 class="text-3xl mb-5">
    2.2.2 Subseries GC Content

</h1>

<p class="mb-3">In this course we will write functions to analyze the GC content of <em>subsequences</em> of a sequence. You can download the sequences right clicking the following link: <a class="link link-hover text-zinc-600" href="data/dengue.fasta">dengue</a> and choosing “Save as…”, saving in choose the <code>Bioinformatics/workspace</code> folder that is in the desktop, as always.</p>
<p class="mb-3">Before starting, load the data:</p>
<div class="sourceCode mb-3 notranslate" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a class="link link-hover text-zinc-600" href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb1-2"><a class="link link-hover text-zinc-600" href="#cb1-2" aria-hidden="true" tabindex="-1"></a>dengue <span class="ot">&lt;-</span> <span class="fu">read.fasta</span>(<span class="st">&quot;dengue.fasta&quot;</span>)[[<span class="dv">1</span>]]</span></code></pre></div>
<h1 class="text-2xl mt-5 mb-5" data-number="1" id="gc-content"><span class="header-section-number">1</span> GC Content</h1>
<p class="mb-3">In the last lecture we defined the GC content of a sequence manually, using the frequencies of <strong>G</strong> and <strong>C</strong> in the sequences. The library <code>seqinr</code> has a function that automatically calculates the GC content for any sequence.</p>
<div class="sourceCode mb-3 notranslate" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a class="link link-hover text-zinc-600" href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">GC</span>(dengue)</span>
<span id="cb2-2"><a class="link link-hover text-zinc-600" href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fl">0.4666977</span></span></code></pre></div>
<p class="mb-3">We will use this function instead today.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="2" id="selecting-windows"><span class="header-section-number">2</span> Selecting windows</h1>
<p class="mb-3">Different parts of the DNA strand behave differently. It is useful to be able to select subparts of it. For example, to select the first 10 nucleotides:</p>
<div class="sourceCode mb-3 notranslate" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a class="link link-hover text-zinc-600" href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> dengue[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb3-2"><a class="link link-hover text-zinc-600" href="#cb3-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">&quot;a&quot;</span> <span class="st">&quot;g&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;g&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;a&quot;</span> <span class="st">&quot;g&quot;</span> <span class="st">&quot;t&quot;</span></span></code></pre></div>
<p class="mb-3">To select <code>n</code> nucleotides, starting from the position <code>s</code>, the operation to be used is <code>dengue[s,(s+n-1)]</code>.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="3" id="choosing-some-windows"><span class="header-section-number">3</span> Choosing some windows</h1>
<p class="mb-3">Let’s analyze the data for two windows of size 100. We will get the GC content for each slice, and save it in an vector.</p>
<div class="sourceCode mb-3 notranslate" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a class="link link-hover text-zinc-600" href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create and empty vector</span></span>
<span id="cb4-2"><a class="link link-hover text-zinc-600" href="#cb4-2" aria-hidden="true" tabindex="-1"></a>gc <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb4-3"><a class="link link-hover text-zinc-600" href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the first element</span></span>
<span id="cb4-4"><a class="link link-hover text-zinc-600" href="#cb4-4" aria-hidden="true" tabindex="-1"></a>gc[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">GC</span>(dengue[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>])</span>
<span id="cb4-5"><a class="link link-hover text-zinc-600" href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the second element</span></span>
<span id="cb4-6"><a class="link link-hover text-zinc-600" href="#cb4-6" aria-hidden="true" tabindex="-1"></a>gc[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">GC</span>(dengue[<span class="dv">101</span><span class="sc">:</span><span class="dv">200</span>])</span></code></pre></div>
<p class="mb-3">Now we have a vector:</p>
<div class="sourceCode mb-3 notranslate" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a class="link link-hover text-zinc-600" href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> gc</span>
<span id="cb5-2"><a class="link link-hover text-zinc-600" href="#cb5-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="fl">0.40</span> <span class="fl">0.48</span></span></code></pre></div>
<p class="mb-3">Which we can plot with <code>barplot(gc)</code>. But how can we do this for all intervals of a given size in a sequence?</p>
<h1 class="text-2xl mt-5 mb-5" data-number="4" id="getting-all-the-intervals"><span class="header-section-number">4</span> Getting all the intervals</h1>
<p class="mb-3">We want to obtain an array such as <code>c(1,101,201,301...)</code>, corresponding to the <strong>start</strong> points for the intervals. but that is too long to write by hand. We will use instead the function <code>seq(start,end,by=size)</code> where <code>start</code> is the first element we want, <code>end</code> the last one, and <code>size</code> the size of the interval between elements.</p>
<p class="mb-3">In the example above, we want:</p>
<div class="sourceCode mb-3 notranslate" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a class="link link-hover text-zinc-600" href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">size</span>(<span class="dv">1</span>,<span class="fu">length</span>(dengue)<span class="sc">-</span><span class="dv">100</span>,<span class="at">by=</span><span class="dv">100</span>)</span></code></pre></div>
<h1 class="text-2xl mt-5 mb-5" data-number="5" id="going-through-all-the-elements"><span class="header-section-number">5</span> Going through all the elements</h1>
<p class="mb-3">To go through all the elements, we will use a <code>for</code> loop. For example, to print all the elements in the vector <code>1:100</code>:</p>
<div class="sourceCode mb-3 notranslate" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a class="link link-hover text-zinc-600" href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>) {</span>
<span id="cb7-2"><a class="link link-hover text-zinc-600" href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb7-3"><a class="link link-hover text-zinc-600" href="#cb7-3" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<h1 class="text-2xl mt-5 mb-5" data-number="6" id="final-result"><span class="header-section-number">6</span> Final result</h1>
<p class="mb-3">Putting everything together, we can write a function that results into a vector containing the GC content of each piece of the sequence.</p>
<div class="sourceCode mb-3 notranslate" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a class="link link-hover text-zinc-600" href="#cb8-1" aria-hidden="true" tabindex="-1"></a>windowGC <span class="ot">&lt;-</span> <span class="cf">function</span>(data, window) {</span>
<span id="cb8-2"><a class="link link-hover text-zinc-600" href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  starts <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="fu">length</span>(data)<span class="sc">-</span>window,<span class="at">by=</span>window)</span>
<span id="cb8-3"><a class="link link-hover text-zinc-600" href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  gc <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb8-4"><a class="link link-hover text-zinc-600" href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(starts)) {</span>
<span id="cb8-5"><a class="link link-hover text-zinc-600" href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> starts[i]</span>
<span id="cb8-6"><a class="link link-hover text-zinc-600" href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> start<span class="sc">+</span>window<span class="dv">-1</span></span>
<span id="cb8-7"><a class="link link-hover text-zinc-600" href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    gc[i] <span class="ot">&lt;-</span> <span class="fu">GC</span>(data[start<span class="sc">:</span>end])</span>
<span id="cb8-8"><a class="link link-hover text-zinc-600" href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-9"><a class="link link-hover text-zinc-600" href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  gc</span>
<span id="cb8-10"><a class="link link-hover text-zinc-600" href="#cb8-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p class="mb-3">To calculate for windows of size 1000:</p>
<div class="sourceCode mb-3 notranslate" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a class="link link-hover text-zinc-600" href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">windowGC</span>(dengue,<span class="dv">1000</span>)</span>
<span id="cb9-2"><a class="link link-hover text-zinc-600" href="#cb9-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="fl">0.477</span> <span class="fl">0.453</span> <span class="fl">0.464</span> <span class="fl">0.441</span> <span class="fl">0.471</span> <span class="fl">0.470</span> <span class="fl">0.488</span> <span class="fl">0.470</span> <span class="fl">0.438</span> <span class="fl">0.471</span></span>
<span id="cb9-3"><a class="link link-hover text-zinc-600" href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">barplot</span>(<span class="fu">windowGC</span>(dengue,<span class="dv">1000</span>))</span></code></pre></div>
<p class="mb-3"><img class="mx-auto" src="data/168364e27ee6e0dc681bf322607d446d.png" /></p>
<p class="mb-3">The function above can be used to compare different window sizes. By decreasing the window size, we can see the GC content with more resolution. Try generating each one of these graphs.</p>
<p class="mb-3"><img class="mx-auto" src="data/c1e1d16948eee263d8373a848e960783.png" /></p>
<p class="mb-3">Looking at this graph, what part of the DNA would break first when heating it?</p>

  <ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul>


</article>
  <script type="text/javascript">
  function googleTranslateElementInit() {
    new google.translate.TranslateElement({
      pageLanguage: 'en',
      autoDisplay: false,
      includedLanguages: 'en,ru,uz',
    }, 'google_translate_element');
  }
  </script>

  <script type="text/javascript" src="https://translate.google.com/translate_a/element.js?cb=googleTranslateElementInit"></script>
</div>

</body>
</html>
