<!DOCTYPE html>
<html lang="en">
<head profile="http://www.w3.org/2005/10/profile">
    <meta charset="UTF-8">
    <meta name="description" content="">
    <meta name="author" content="">
    <meta name=viewport content="width=device-width, initial-scale=1">
    <title>2.3.2 Codon frequencies and finding genes
</title>
    <link rel="stylesheet" href="/css/style.css">
    <link rel="stylesheet" href="/css/syntax.css">
    <link href="https://cdn.jsdelivr.net/npm/daisyui@3.1.0/dist/full.css" rel="stylesheet" type="text/css" />
    <script src="https://cdn.tailwindcss.com"></script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.css" integrity="sha384-GvrOXuhMATgEsSwCs4smul74iXGOixntILdUW9XmUC6+HX0sLNAK3q71HotJqlAn" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.js" integrity="sha384-cpW21h6RZv/phavutF+AuVYrr+dA8xD9zs6FwLpaCct6O9ctzYFfFr4dgmgccOTx" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/contrib/auto-render.min.js" integrity="sha384-+VBxd3r6XgURycqtZ117nYw44OOcIax56Z4dCRWbxyPt0Koah1uHoK0o4+/RRE05" crossorigin="anonymous"
        onload="renderMathInElement(document.body);"></script>

</head>
<body>
    <header>
        <div class="navbar bg-base-100">
          <div class="navbar-start">
            <nav>
            <ol class="inline-flex items-center space-x-1 md:space-x-3">
        <li class="inline-flex items-center">
          <a href="/" class="inline-flex items-center text-l font-medium text-gray-700 hover:text-blue-600 dark:text-gray-400 dark:hover:text-white">
            <svg aria-hidden="true" class="w-4 h-4 mr-2" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"></path></svg>
            Juan Raphael Diaz Simões
          </a>
        </li>
    
        <li>
      <div class="flex items-center">
        <svg aria-hidden="true" class="w-6 h-6 text-gray-400" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" clip-rule="evenodd"></path></svg>
        <a href="/courses" class="ml-1 text-l font-medium text-gray-700 hover:text-blue-600 md:ml-2 dark:text-gray-400 dark:hover:text-white">Courses</a>
      </div>
    </li>

          </div>
          <div class="navbar-end">
            <ul class="menu menu-horizontal px-1">
              <li><a href="/posts">Posts</a></li>
              <li><a href="/courses">Courses</a></li>
              <li><a href="/cv.html">CV</a></li>
              <li><a href="/projects.html">Projects</a></li>
            </ul>
          </div>
        </nav>
    </header>
    <div class="flex justify-center bg-base-100">
<article class="prose max-w-prose mb-5">
  <div class="book-navigation">

    <div class="book-navigation-parent drawer">
      <input id="my-drawer" type="checkbox" class="drawer-toggle" />
      <div class="drawer-content">
        <label for="my-drawer">
          <div class="space-y-1 inline-block">
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
          </div>
          <b>Index</b>
        </label>
      </div>
      <div class="drawer-side">
        <label for="my-drawer" class="drawer-overlay"></label>
           <div class="menu menu-vertical p-4 bg-base-200">
             <h1 class="text-xl mb-5 w-100" style="margin-left: 0.75em;">
                  Introduction to Bioinformatics

             </h1>
             <div style="margin-left:-1.5em;">
               <ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/index.html">1 Lectures
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/03-lecture-01/index.html">1.1 Bioinformatics Basics
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/07-lecture-02/index.html">1.2 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/10-lecture-03/index.html">1.3 Comparing Gene Sequences 2
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/13-lecture-04/index.html">1.4 Gene Finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/16-lecture-05/index.html">1.5 Locating genes and its functions: techniques, problems and difficulties
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/index.html">2 Practical Lessons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/index.html">2.1 Basic R and generic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/01-seminar-01/index.html">2.1.1 Basic R
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/02-seminar-02/index.html">2.1.2 Files
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/04-seminar-03/index.html">2.1.3 Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/05-seminar-04/index.html">2.1.4 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/06-control-01/index.html">2.1.5 Control: Loading and processing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/20-seminar-12/index.html">2.1.6 Programming Exercises
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/index.html">2.2 Basic analysis of genetic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/08-seminar-05/index.html">2.2.1 GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/09-seminar-06/index.html">2.2.2 Subseries GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/10-seminar-06/index.html">2.2.3 Exercise: Comparing sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/11-seminar-07/index.html">2.2.4 Creating dot plots
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/14-control-02/index.html">2.2.5 Loading and comparing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/index.html">2.3 Codons and gene finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/15-seminar-09/index.html">2.3.1 Translating bases to codons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/17-seminar-10/index.html">2.3.2 Codon frequencies and finding genes
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/18-control-03/index.html">2.3.3 Theoretical Control
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/19-seminar-11/index.html">2.3.4 Aminoacid sequences and base sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/index.html">2.4 Summary and exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/21-seminar-13/index.html">2.4.1 Theoretical Questions
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/22-exam-01/index.html">2.4.2 Final exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li></ul></li></ul>
             </div>
           </div>
      </div>
    </div>

    <div class="book-navigation-parent">
      <div class="arrow arrow-up"></div>
      <a href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/index.html">2.3 Codons and gene finding
</a>
      <div class="arrow arrow-up"></div>
    </div>

    <div class="book-navigation-section">

       <div class="book-navigation-previous">
         <div class="arrow arrow-left"></div>
         <a href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/15-seminar-09/index.html">2.3.1 Translating bases to codons
</a>
       </div>

       <div class="book-navigation-next">
         <a href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/18-control-03/index.html">2.3.3 Theoretical Control
</a>
         <div class="arrow arrow-right"></div>
       </div>
    </div>

    <div id="google_translate_element" style="margin-top:1em;"></div>

  </div>

    <h1 class="text-3xl mb-5">
    2.3.2 Codon frequencies and finding genes

</h1>

<h1 class="text-2xl mt-5 mb-5" data-number="1" id="introduction"><span class="header-section-number">1</span> Introduction</h1>
<p class="mb-3">Today we will be studying codon frequencies, and automatically find genes in DNA sequences. You can download the sequences by clicking the following links: <a class="link link-hover text-zinc-600" href="data/dengue.fasta">dengue</a> and <a class="link link-hover text-zinc-600" href="data/zika.fasta">zika</a>. Right click it, and choose “Save as…” and choose the <code>Bioinformatics/workspace</code> folder that is in the desktop, as always.</p>
<p class="mb-3">After, copy and paste the following code in the RStudio console, which we will use during the class.</p>
<pre class="sourcecode"><code>library(seqinr)
library(Biostrings)

# Load vector of nucleotides from file
readDNAString &lt;- function(file) {
  DNAString(toupper(c2s(read.fasta(file)[[1]])))
}

# Start and end codons
start = &quot;M&quot;
end = &quot;*&quot;

# Translate a sequence of bases into codons
translateAndRegularize &lt;- function(data, index) {
  l &lt;- (length(data)- index+1) %/% 3 * 3
  reg &lt;- data[index:(l+index-1)]
  Biostrings::translate(reg,no.init.codon = TRUE)
}

# Get first element satisfying constraint
firstCompatible &lt;- function(s, es) {
  if(length(es) &gt;= 1) {
    for (i in 1:length(es)) {
      e &lt;- es[i]
      if(e &gt; s) {
        return(e)
      }
    }
  }
  return(NA)
}

# Get possible genes for a sequence
getPossibleGenes &lt;- function(dna) {
  starts &lt;- Biostrings::matchPattern(start, dna)@ranges@start
  ends &lt;- Biostrings::matchPattern(end, dna)@ranges@start
  sequences &lt;- c()
  for(i in 1:length(starts)) {
    startEnd &lt;- firstCompatible(starts[i], ends)
    if(!is.na(startEnd)) {
      sequences &lt;- c(sequences,dna[starts[i]:startEnd])
    }
  }
  sequences
}

# Get all possible genes, using the three frames
getAllPossibleGenes &lt;- function(data) {
  l1 &lt;- getPossibleGenes(translateAndRegularize(data,1))
  l2 &lt;- getPossibleGenes(translateAndRegularize(data,2))
  l3 &lt;- getPossibleGenes(translateAndRegularize(data,3))
  gs &lt;- c(l1,l2,l3)
  len &lt;- sapply(gs,length)
  gs[order(len)]
}

# Plot frequencies
frequencies &lt;- function(data) {
  barplot(table(strsplit(as.character(data),split=&quot;&quot;)),cex.names=0.6)
}</code></pre>
<p class="mb-3">After that, load the dengue and zika sequences:</p>
<pre class="sourcecode"><code>dengue &lt;- readDNAString(&quot;dengue.fasta&quot;)
zika &lt;- readDNAString(&quot;zika.fasta&quot;)</code></pre>
<h1 class="text-2xl mt-5 mb-5" data-number="2" id="codon-frequencies"><span class="header-section-number">2</span> Codon frequencies</h1>
<p class="mb-3">Previously, we studied base sequences:</p>
<pre class="sourcecode"><code>frequencies(dengue)</code></pre>
<p class="mb-3"><img class="mx-auto" src="data/a68fd58e4f48c3e3fd4128f17060645f.png" /></p>
<p class="mb-3">The same can be done for codons. However, there are three possible positions for codons, so we need to graph these thee:</p>
<pre class="sourcecode"><code>&gt; frequencies(translateAndRegularize(dengue,1))
&gt; frequencies(translateAndRegularize(dengue,2))
&gt; frequencies(translateAndRegularize(dengue,3))</code></pre>
<p class="mb-3">The first one looks like this:</p>
<p class="mb-3"><img class="mx-auto" src="data/dabc782d5dcf3304d8b1cb57f3c5fd95.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="2.1" id="questions"><span class="header-section-number">2.1</span> Questions</h2>
<ol class="list-decimal list-outside mt-1 mb-3 pl-5" type="1">
<li>Look at the graphs for positions 1, 2 and 3. Do they look the same?</li>
<li>Do the same four graphs (one bases and three for codons) for the zika virus.</li>
</ol>
<h1 class="text-2xl mt-5 mb-5" data-number="3" id="gene-finding"><span class="header-section-number">3</span> Gene finding</h1>
<p class="mb-3">In the previous classes we learned about the simple method for looking for genes in a sequence. This method is implemented in the <code>getAllPossibleGenes</code> function. For example:</p>
<pre class="sourcecode"><code>&gt; getAllPossibleGenes(DNAString(&quot;ATGTAG&quot;))
[[1]]
2-letter AAString object
seq: M*</code></pre>
<p class="mb-3">The advantage of this function is that we can do that for larger sequences:</p>
<pre class="sourcecode"><code>&gt; dengueGenes &lt;- getAllPossibleGenes(dengue)
&gt; dengueGenes</code></pre>
<p class="mb-3">Each identified sequence has a different size, and as discussed during the letter, this is one of the main characteristics used to discern between false and true genes. The resulting list is sorted, look at the sizes for each sequence.</p>
<p class="mb-3">One can plot sizes with:</p>
<pre class="sourcecode"><code>hist(sapply(dengueGenes,length),100,main=&quot;ORF Lengths&quot;,xlab=&quot;Length&quot;)</code></pre>
<p class="mb-3"><img class="mx-auto" src="data/28c5367cf8cefe40cee94449d26e574b.png" /></p>
<p class="mb-3">This plot makes it simpler to draw a boundary between sequences that are two small and those who have a sufficient size.</p>
<h2 class="text-xl mt-5 mb-5" data-number="3.1" id="questions-1"><span class="header-section-number">3.1</span> Questions</h2>
<ol class="list-decimal list-outside mt-1 mb-3 pl-5" type="1">
<li>How many sequences have a size bigger than 100? Show <code>dengueGenes</code> and take a look at array indices.</li>
<li>Repeat this analysis for the zika virus.</li>
</ol>

  <ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul>


</article>
  <script type="text/javascript">
  function googleTranslateElementInit() {
    new google.translate.TranslateElement({
      pageLanguage: 'en',
      autoDisplay: false,
      includedLanguages: 'en,ru,uz',
    }, 'google_translate_element');
  }
  </script>

  <script type="text/javascript" src="https://translate.google.com/translate_a/element.js?cb=googleTranslateElementInit"></script>
</div>

</body>
</html>
