<!DOCTYPE html>
<html lang="en">
<head profile="http://www.w3.org/2005/10/profile">
    <meta charset="UTF-8">
    <meta name="description" content="">
    <meta name="author" content="">
    <meta name=viewport content="width=device-width, initial-scale=1">
    <title>2.1.4 Comparing Gene Sequences
</title>
    <link rel="stylesheet" href="/css/style.css">
    <link rel="stylesheet" href="/css/syntax.css">
    <link href="https://cdn.jsdelivr.net/npm/daisyui@3.1.0/dist/full.css" rel="stylesheet" type="text/css" />
    <script src="https://cdn.tailwindcss.com"></script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.css" integrity="sha384-GvrOXuhMATgEsSwCs4smul74iXGOixntILdUW9XmUC6+HX0sLNAK3q71HotJqlAn" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.js" integrity="sha384-cpW21h6RZv/phavutF+AuVYrr+dA8xD9zs6FwLpaCct6O9ctzYFfFr4dgmgccOTx" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/contrib/auto-render.min.js" integrity="sha384-+VBxd3r6XgURycqtZ117nYw44OOcIax56Z4dCRWbxyPt0Koah1uHoK0o4+/RRE05" crossorigin="anonymous"
        onload="renderMathInElement(document.body);"></script>

</head>
<body>
    <header>
        <div class="navbar bg-base-100">
          <div class="navbar-start">
            <nav>
            <ol class="inline-flex items-center space-x-1 md:space-x-3">
        <li class="inline-flex items-center">
          <a href="/" class="inline-flex items-center text-l font-medium text-gray-700 hover:text-blue-600 dark:text-gray-400 dark:hover:text-white">
            <svg aria-hidden="true" class="w-4 h-4 mr-2" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"></path></svg>
            Juan Raphael Diaz Sim√µes
          </a>
        </li>
    
        <li>
      <div class="flex items-center">
        <svg aria-hidden="true" class="w-6 h-6 text-gray-400" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" clip-rule="evenodd"></path></svg>
        <a href="/courses" class="ml-1 text-l font-medium text-gray-700 hover:text-blue-600 md:ml-2 dark:text-gray-400 dark:hover:text-white">Courses</a>
      </div>
    </li>

          </div>
          <div class="navbar-end">
            <ul class="menu menu-horizontal px-1">
              <li><a href="/posts">Posts</a></li>
              <li><a href="/courses">Courses</a></li>
              <li><a href="/cv.html">CV</a></li>
              <li><a href="/projects.html">Projects</a></li>
            </ul>
          </div>
        </nav>
    </header>
    <div class="flex justify-center bg-base-100">
<article class="prose max-w-prose mb-5">
  <div class="book-navigation">

    <div class="book-navigation-parent drawer">
      <input id="my-drawer" type="checkbox" class="drawer-toggle" />
      <div class="drawer-content">
        <label for="my-drawer">
          <div class="space-y-1 inline-block">
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
          </div>
          <b>Index</b>
        </label>
      </div>
      <div class="drawer-side">
        <label for="my-drawer" class="drawer-overlay"></label>
           <div class="menu menu-vertical p-4 bg-base-200">
             <h1 class="text-xl mb-5 w-100" style="margin-left: 0.75em;">
                  Introduction to Bioinformatics

             </h1>
             <div style="margin-left:-1.5em;">
               <ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/index.html">1 Lectures
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/03-lecture-01/index.html">1.1 Bioinformatics Basics
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/07-lecture-02/index.html">1.2 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/10-lecture-03/index.html">1.3 Comparing Gene Sequences 2
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/13-lecture-04/index.html">1.4 Gene Finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/16-lecture-05/index.html">1.5 Locating genes and its functions: techniques, problems and difficulties
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/index.html">2 Practical Lessons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/index.html">2.1 Basic R and generic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/01-seminar-01/index.html">2.1.1 Basic R
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/02-seminar-02/index.html">2.1.2 Files
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/04-seminar-03/index.html">2.1.3 Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/05-seminar-04/index.html">2.1.4 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/06-control-01/index.html">2.1.5 Loading and processing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/20-seminar-12/index.html">2.1.6 Programming Exercises
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/index.html">2.2 Basic analysis of genetic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/08-seminar-05/index.html">2.2.1 GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/09-seminar-06/index.html">2.2.2 Subseries GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/11-seminar-07/index.html">2.2.3 Creating dot plots
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/14-control-02/index.html">2.2.4 Loading and comparing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/index.html">2.3 Codons and gene finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/15-seminar-09/index.html">2.3.1 Translating bases to codons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/17-seminar-10/index.html">2.3.2 Codon frequencies and finding genes
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/18-control-03/index.html">2.3.3 Theoretical Control
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/19-seminar-11/index.html">2.3.4 Aminoacid sequences and base sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/index.html">2.4 Summary and exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/21-seminar-13/index.html">2.4.1 Theoretical Questions
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/22-exam-01/index.html">2.4.2 Final exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li></ul></li></ul>
             </div>
           </div>
      </div>
    </div>

    <div class="book-navigation-parent">
      <div class="arrow arrow-up"></div>
      <a href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/index.html">2.1 Basic R and generic data
</a>
      <div class="arrow arrow-up"></div>
    </div>

    <div class="book-navigation-section">

       <div class="book-navigation-previous">
         <div class="arrow arrow-left"></div>
         <a href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/04-seminar-03/index.html">2.1.3 Gene Sequences
</a>
       </div>

       <div class="book-navigation-next">
         <a href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/06-control-01/index.html">2.1.5 Loading and processing data
</a>
         <div class="arrow arrow-right"></div>
       </div>
    </div>

    <div id="google_translate_element" style="margin-top:1em;"></div>

  </div>

    <h1 class="text-3xl mb-5">
    2.1.4 Comparing Gene Sequences

</h1>

<p class="mb-3">In the last lecture, we fetched data from <a class="link link-hover text-zinc-600" href="www.ncbi.nlm.nih.gov">NCBI</a> and did some simple statistical analysis on the dengue virus genome. We would like to do such analyses very easily with other genomes. For this, we are going to use <em>functions</em>.</p>
<p class="mb-3">Furthermore, we will use these new functions two analyze two different viruses: the dengue virus (<code>NC_001477</code>) and the zika virus (<code>NC_012532</code>). Both are viruses that cause diseases in tropical regions and are transmitted by mosquitos, so it is interesting to see if we can find simple differences between the genomes.</p>
<p class="mb-3">You can download the sequences by clicking the following links: <a class="link link-hover text-zinc-600" href="data/dengue.fasta">dengue</a> and <a class="link link-hover text-zinc-600" href="data/zika.fasta">zika</a>. It will be easier if you right click it, and choose ‚ÄúSave as‚Ä¶‚Äù and choose the <code>Bioinformatics/workspace</code> folder that is in the desktop.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="1" id="functions"><span class="header-section-number">1</span> Functions</h1>
<p class="mb-3">Functions are operations that are composed by smaller operations. They are useful for putting together tasks that are always executed together, without being tied to a specific data.</p>
<p class="mb-3">For example, let‚Äôs say we want to print a word together with its number of characters. This can be done as follows:</p>
<div class="sourceCode mb-3 notranslate" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a class="link link-hover text-zinc-600" href="#cb1-1" aria-hidden="true" tabindex="-1"></a>word <span class="ot">&lt;-</span> <span class="st">&quot;Juan&quot;</span></span>
<span id="cb1-2"><a class="link link-hover text-zinc-600" href="#cb1-2" aria-hidden="true" tabindex="-1"></a>size <span class="ot">&lt;-</span> <span class="fu">nchar</span>(<span class="st">&quot;Juan&quot;</span>)</span>
<span id="cb1-3"><a class="link link-hover text-zinc-600" href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(word,<span class="st">&quot;has&quot;</span>,size,<span class="st">&quot;letters.&quot;</span>)</span></code></pre></div>
<p class="mb-3">The result will be <code>Juan has 4 letters.</code> What if we want to execute this task for any word? We do this by defining a function:</p>
<div class="sourceCode mb-3 notranslate" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a class="link link-hover text-zinc-600" href="#cb2-1" aria-hidden="true" tabindex="-1"></a>countSize <span class="ot">&lt;-</span> <span class="cf">function</span>(word) {</span>
<span id="cb2-2"><a class="link link-hover text-zinc-600" href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  size <span class="ot">&lt;-</span> <span class="fu">nchar</span>(word)</span>
<span id="cb2-3"><a class="link link-hover text-zinc-600" href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(word,<span class="st">&quot;has&quot;</span>,size,<span class="st">&quot;letters.&quot;</span>)</span>
<span id="cb2-4"><a class="link link-hover text-zinc-600" href="#cb2-4" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p class="mb-3">and it can be used with different words:</p>
<div class="sourceCode mb-3 notranslate" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a class="link link-hover text-zinc-600" href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">countSize</span>(<span class="st">&quot;Juan&quot;</span>)</span>
<span id="cb3-2"><a class="link link-hover text-zinc-600" href="#cb3-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">&quot;Juan has 4 letters.&quot;</span></span>
<span id="cb3-3"><a class="link link-hover text-zinc-600" href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">countSize</span>(<span class="st">&quot;Dengue&quot;</span>)</span>
<span id="cb3-4"><a class="link link-hover text-zinc-600" href="#cb3-4" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">&quot;Dengue has 6 letters.&quot;</span></span></code></pre></div>
<p class="mb-3">We can apply this method for different operations.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="2" id="loading-gene-sequence-data"><span class="header-section-number">2</span> Loading gene sequence data</h1>
<p class="mb-3">The idea for this function is to make the file we are reading a function argument. Based on this file, we will build the vector data. Text that are after <code>#</code> are just comments and have no effect on the code.</p>
<div class="sourceCode mb-3 notranslate" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a class="link link-hover text-zinc-600" href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load vector of nucleotides from file</span></span>
<span id="cb4-2"><a class="link link-hover text-zinc-600" href="#cb4-2" aria-hidden="true" tabindex="-1"></a>readVector <span class="ot">&lt;-</span> <span class="cf">function</span>(file){</span>
<span id="cb4-3"><a class="link link-hover text-zinc-600" href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read data from file</span></span>
<span id="cb4-4"><a class="link link-hover text-zinc-600" href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.fasta</span>(file)</span>
<span id="cb4-5"><a class="link link-hover text-zinc-600" href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Transform data into just a vector</span></span>
<span id="cb4-6"><a class="link link-hover text-zinc-600" href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  vector <span class="ot">&lt;-</span> <span class="fu">c</span>(data[[<span class="dv">1</span>]])</span>
<span id="cb4-7"><a class="link link-hover text-zinc-600" href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return vector</span></span>
<span id="cb4-8"><a class="link link-hover text-zinc-600" href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(vector)</span>
<span id="cb4-9"><a class="link link-hover text-zinc-600" href="#cb4-9" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p class="mb-3">This can be used to load the zika and dengue data we downloaded before:</p>
<div class="sourceCode mb-3 notranslate" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a class="link link-hover text-zinc-600" href="#cb5-1" aria-hidden="true" tabindex="-1"></a>dengue <span class="ot">&lt;-</span> <span class="fu">readVector</span>(<span class="st">&quot;dengue.fasta&quot;</span>)</span>
<span id="cb5-2"><a class="link link-hover text-zinc-600" href="#cb5-2" aria-hidden="true" tabindex="-1"></a>zika <span class="ot">&lt;-</span> <span class="fu">readVector</span>(<span class="st">&quot;zika.fasta&quot;</span>)</span></code></pre></div>
<p class="mb-3">As a first test, we can compare their sizes:</p>
<div class="sourceCode mb-3 notranslate" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a class="link link-hover text-zinc-600" href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">length</span>(dengue)</span>
<span id="cb6-2"><a class="link link-hover text-zinc-600" href="#cb6-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">10735</span></span>
<span id="cb6-3"><a class="link link-hover text-zinc-600" href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">length</span>(zika)</span>
<span id="cb6-4"><a class="link link-hover text-zinc-600" href="#cb6-4" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">10794</span></span></code></pre></div>
<p class="mb-3">their sizes are different, but not too far away from each other.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="3" id="showing-data"><span class="header-section-number">3</span> Showing data</h1>
<p class="mb-3">Last time, we built tables from this data, and plotted a bar plot with them. This can also become a function:</p>
<div class="sourceCode mb-3 notranslate" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a class="link link-hover text-zinc-600" href="#cb7-1" aria-hidden="true" tabindex="-1"></a>showBars <span class="ot">&lt;-</span> <span class="cf">function</span>(title,data, n) {</span>
<span id="cb7-2"><a class="link link-hover text-zinc-600" href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  table <span class="ot">&lt;-</span> <span class="fu">count</span>(data,n)</span>
<span id="cb7-3"><a class="link link-hover text-zinc-600" href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">barplot</span>(table,<span class="at">main=</span>title)</span>
<span id="cb7-4"><a class="link link-hover text-zinc-600" href="#cb7-4" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p class="mb-3">And we can see the bar plots for each distribution:</p>
<div class="sourceCode mb-3 notranslate" id="cb8"><pre class="sourceCode javascript"><code class="sourceCode javascript"><span id="cb8-1"><a class="link link-hover text-zinc-600" href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">showBars</span>(<span class="st">&quot;Dengue&quot;</span><span class="op">,</span>dengue<span class="op">,</span><span class="dv">1</span>)</span>
<span id="cb8-2"><a class="link link-hover text-zinc-600" href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">showBars</span>(<span class="st">&quot;Zika&quot;</span><span class="op">,</span>zika<span class="op">,</span><span class="dv">1</span>)</span></code></pre></div>
<h1 class="text-2xl mt-5 mb-5" data-number="4" id="comparing-data"><span class="header-section-number">4</span> Comparing data</h1>
<p class="mb-3">These plots look different. However, we are just showing them one at time. How can we show both at the same time?</p>
<p class="mb-3">First, we tell R that we want to see data in a table with 1 row and 2 columns:</p>
<div class="sourceCode mb-3 notranslate" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a class="link link-hover text-zinc-600" href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span></code></pre></div>
<p class="mb-3">and then, we plot once more our graphs:</p>
<div class="sourceCode mb-3 notranslate" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a class="link link-hover text-zinc-600" href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">showBars</span>(<span class="st">&quot;Dengue&quot;</span>,dengue,<span class="dv">1</span>)</span>
<span id="cb10-2"><a class="link link-hover text-zinc-600" href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">showBars</span>(<span class="st">&quot;Zika&quot;</span>,zika,<span class="dv">1</span>)</span></code></pre></div>
<p class="mb-3"><img class="mx-auto" src="data/comparison.png" /></p>
<p class="mb-3">These graphs have a similar shape, but the quantities of each nucleotide are different. How can we measure this difference? We can write another function.</p>
<div class="sourceCode mb-3 notranslate" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a class="link link-hover text-zinc-600" href="#cb11-1" aria-hidden="true" tabindex="-1"></a>difference <span class="ot">&lt;-</span> <span class="cf">function</span>(data1,data2,n) {</span>
<span id="cb11-2"><a class="link link-hover text-zinc-600" href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  table1 <span class="ot">&lt;-</span> <span class="fu">count</span>(data1,n)</span>
<span id="cb11-3"><a class="link link-hover text-zinc-600" href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  table2 <span class="ot">&lt;-</span> <span class="fu">count</span>(data2,n)</span>
<span id="cb11-4"><a class="link link-hover text-zinc-600" href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">barplot</span>(table1<span class="sc">-</span>table2,<span class="at">main=</span><span class="st">&quot;Difference&quot;</span>)</span>
<span id="cb11-5"><a class="link link-hover text-zinc-600" href="#cb11-5" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p class="mb-3">and execute it with <code>difference(dengue,zika,1)</code>:</p>
<p class="mb-3"><img class="mx-auto" src="data/difference.png" /> As one can see, there are more <code>A</code> nucleotides in the dengue genome, but less <code>G</code> ones.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="5" id="exercise"><span class="header-section-number">5</span> Exercise</h1>
<p class="mb-3">Try to show the bar plots of dengue and zika using sequences of size 2 instead. What are the differences?</p>

  <ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul>


</article>
  <script type="text/javascript">
  function googleTranslateElementInit() {
    new google.translate.TranslateElement({
      pageLanguage: 'en',
      autoDisplay: false,
      includedLanguages: 'en,ru,uz',
    }, 'google_translate_element');
  }
  </script>

  <script type="text/javascript" src="https://translate.google.com/translate_a/element.js?cb=googleTranslateElementInit"></script>
</div>

</body>
</html>
