<!DOCTYPE html>
<html lang="en">
<head profile="http://www.w3.org/2005/10/profile">
    <meta charset="UTF-8">
    <meta name="description" content="">
    <meta name="author" content="">
    <meta name=viewport content="width=device-width, initial-scale=1">
    <title>2.1.6 Programming Exercises
</title>
    <link rel="stylesheet" href="/css/style.css">
    <link rel="stylesheet" href="/css/syntax.css">
    <link href="https://cdn.jsdelivr.net/npm/daisyui@3.1.0/dist/full.css" rel="stylesheet" type="text/css" />
    <script src="https://cdn.tailwindcss.com"></script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.css" integrity="sha384-GvrOXuhMATgEsSwCs4smul74iXGOixntILdUW9XmUC6+HX0sLNAK3q71HotJqlAn" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.js" integrity="sha384-cpW21h6RZv/phavutF+AuVYrr+dA8xD9zs6FwLpaCct6O9ctzYFfFr4dgmgccOTx" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/contrib/auto-render.min.js" integrity="sha384-+VBxd3r6XgURycqtZ117nYw44OOcIax56Z4dCRWbxyPt0Koah1uHoK0o4+/RRE05" crossorigin="anonymous"
        onload="renderMathInElement(document.body);"></script>

</head>
<body>
    <header>
        <div class="navbar bg-base-100">
          <div class="navbar-start">
            <nav>
            <ol class="inline-flex items-center space-x-1 md:space-x-3">
        <li class="inline-flex items-center">
          <a href="/" class="inline-flex items-center text-l font-medium text-gray-700 hover:text-blue-600 dark:text-gray-400 dark:hover:text-white">
            <svg aria-hidden="true" class="w-4 h-4 mr-2" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"></path></svg>
            Juan Raphael Diaz Sim√µes
          </a>
        </li>
    
        <li>
      <div class="flex items-center">
        <svg aria-hidden="true" class="w-6 h-6 text-gray-400" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" clip-rule="evenodd"></path></svg>
        <a href="/courses" class="ml-1 text-l font-medium text-gray-700 hover:text-blue-600 md:ml-2 dark:text-gray-400 dark:hover:text-white">Courses</a>
      </div>
    </li>

          </div>
          <div class="navbar-end">
            <ul class="menu menu-horizontal px-1">
              <li><a href="/posts">Posts</a></li>
              <li><a href="/courses">Courses</a></li>
              <li><a href="/cv.html">CV</a></li>
              <li><a href="/projects.html">Projects</a></li>
            </ul>
          </div>
        </nav>
    </header>
    <div class="flex justify-center bg-base-100">
<article class="prose max-w-prose mb-5">
  <div class="book-navigation">

    <div class="book-navigation-parent drawer">
      <input id="my-drawer" type="checkbox" class="drawer-toggle" />
      <div class="drawer-content">
        <label for="my-drawer">
          <div class="space-y-1 inline-block">
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
          </div>
          <b>Index</b>
        </label>
      </div>
      <div class="drawer-side">
        <label for="my-drawer" class="drawer-overlay"></label>
           <div class="menu menu-vertical p-4 bg-base-200">
             <h1 class="text-xl mb-5 w-100" style="margin-left: 0.75em;">
                  Introduction to Bioinformatics

             </h1>
             <div style="margin-left:-1.5em;">
               <ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/index.html">1 Lectures
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/03-lecture-01/index.html">1.1 Bioinformatics Basics
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/07-lecture-02/index.html">1.2 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/10-lecture-03/index.html">1.3 Comparing Gene Sequences 2
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/13-lecture-04/index.html">1.4 Gene Finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/16-lecture-05/index.html">1.5 Locating genes and its functions: techniques, problems and difficulties
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/index.html">2 Practical Lessons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/index.html">2.1 Basic R and generic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/01-seminar-01/index.html">2.1.1 Basic R
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/02-seminar-02/index.html">2.1.2 Files
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/04-seminar-03/index.html">2.1.3 Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/05-seminar-04/index.html">2.1.4 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/06-control-01/index.html">2.1.5 Control: Loading and processing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/20-seminar-12/index.html">2.1.6 Programming Exercises
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/index.html">2.2 Basic analysis of genetic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/08-seminar-05/index.html">2.2.1 GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/09-seminar-06/index.html">2.2.2 Subseries GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/10-seminar-06/index.html">2.2.3 Exercise: Comparing sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/11-seminar-07/index.html">2.2.4 Creating dot plots
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/14-control-02/index.html">2.2.5 Loading and comparing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/index.html">2.3 Codons and gene finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/15-seminar-09/index.html">2.3.1 Translating bases to codons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/17-seminar-10/index.html">2.3.2 Codon frequencies and finding genes
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/18-control-03/index.html">2.3.3 Theoretical Control
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/19-seminar-11/index.html">2.3.4 Aminoacid sequences and base sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/index.html">2.4 Summary and exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/21-seminar-13/index.html">2.4.1 Theoretical Questions
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/22-exam-01/index.html">2.4.2 Final exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li></ul></li></ul>
             </div>
           </div>
      </div>
    </div>

    <div class="book-navigation-parent">
      <div class="arrow arrow-up"></div>
      <a href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/index.html">2.1 Basic R and generic data
</a>
      <div class="arrow arrow-up"></div>
    </div>

    <div class="book-navigation-section">

       <div class="book-navigation-previous">
         <div class="arrow arrow-left"></div>
         <a href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/06-control-01/index.html">2.1.5 Control: Loading and processing data
</a>
       </div>

       <div class="book-navigation-next">
       </div>
    </div>

    <div id="google_translate_element" style="margin-top:1em;"></div>

  </div>

    <h1 class="text-3xl mb-5">
    2.1.6 Programming Exercises

</h1>

<p class="mb-3">In this lesson, you will solve, by yourselves, the following four questions. They are purely about R, to complement the theoretical exercise from two weeks ago. All the information necessary to solve it is below, in the ‚ÄúBasic R‚Äù section.</p>
<p class="mb-3">Open RStudio, write the solutions in the text editor panel (top left), and show me when each one is ready.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="1" id="exercises"><span class="header-section-number">1</span> Exercises</h1>
<ol class="list-decimal list-outside mt-1 mb-3 pl-5" type="1">
<li><p class="mb-3">Sum all the numbers from 1 to 1000. You can do it using the syntax <code>1:1000</code>, which generates a vector with all numbers between 1 and 1000, and the function <code>sum</code>. Do the same with the function <code>mean</code>, which calculates the mean.</p></li>
<li><p class="mb-3">Write a function called <code>sayHi</code> that takes a parameter <code>name</code>, and returns the sentence <code>Hi name!</code>. For this, you will use the function <code>paste</code>, which can be used as follows:</p>
<pre class="sourcecode"><code>&gt; paste(&quot;A&quot;,&quot;B&quot;)
[1] &quot;A B&quot;
&gt; paste(&quot;A&quot;,&quot;B&quot;,sep=&quot;&quot;)
[1] &quot;AB&quot;</code></pre></li>
<li><p class="mb-3">Create a vector with the name of four of your classmates, using the <code>c(...)</code> construction. Use a <code>for</code> loop to go through these names, an print the say hi sentence for each one of them, using the <code>print</code> function, and the <code>sayHi</code> function that was defined just above.</p></li>
<li><p class="mb-3">Write a function that takes a vector <code>numbers</code> as argument, and for each one tells if the number is big (&gt;10) or small, using the <code>if/else</code> construction presented in the <em>Conditionals</em> part.</p></li>
</ol>
<h1 class="text-2xl mt-5 mb-5" data-number="2" id="basic-r"><span class="header-section-number">2</span> Basic R</h1>
<p class="mb-3">This section serves as a reference on how to use the most useful constructs in R. Make sure to come back to this page if you forget the syntax of the operations.</p>
<h2 class="text-xl mt-5 mb-5" data-number="2.1" id="variables"><span class="header-section-number">2.1</span> Variables</h2>
<p class="mb-3">R can act like a basic calculator that you type commands in to. You can also use it like a more advanced scientific calculator and create <em>variables</em> that store information. In R, we assign values to variables using an arrow-looking function <code>&lt;-</code> the <em>assignment operator</em>. For example, we can assign the value <code>2*3</code> to the variable <code>x</code> using the command:</p>
<pre class="sourcecode"><code>&gt; x &lt;- 2*3</code></pre>
<p class="mb-3">To view the contents of any variable, just type its name, press enter, and the contents of that R object will be displayed:</p>
<pre class="sourcecode"><code>&gt; x
[1] 6</code></pre>
<p class="mb-3">What kind of information can you store in a variable? Anything! But the simplest kinds of data are strings, numbers and booleans:</p>
<pre class="sourcecode"><code>&gt; s &lt;- &quot;Some String&quot;
&gt; n &lt;- 154.4
&gt; b &lt;- TRUE</code></pre>
<h2 class="text-xl mt-5 mb-5" data-number="2.2" id="vectors"><span class="header-section-number">2.2</span> Vectors</h2>
<p class="mb-3">To create a vector, we can use the <code>c()</code> (combine) function. For example, to create a vector called <code>myvector</code> that has elements with values 8, 6, 9, 10, and 5, we type:</p>
<pre class="sourcecode"><code>myvector &lt;- c(8, 6, 9, 10, 5) # note: commas between each number!</code></pre>
<p class="mb-3">To see the contents of the variable <code>myvector</code>, we can just type its name and press enter:</p>
<pre class="sourcecode"><code>&gt; myvector
[1]  8  6  9 10  5</code></pre>
<p class="mb-3">The <code>[1]</code> is the <em>index</em> of the first <em>element</em> in the vector. We can extract any element of the vector by typing the vector name with the index of that element given in <em>square brackets</em> <code>[...]</code>.</p>
<p class="mb-3">For example, to get the value of the 4th element in the vector <code>myvector</code>, we type:</p>
<pre class="sourcecode"><code>&gt; myvector[4]
[1] 10</code></pre>
<p class="mb-3">Vectors can also contain letters, such as those designating nucleic acids</p>
<pre class="sourcecode"><code>my.seq &lt;- c(&quot;A&quot;,&quot;T&quot;,&quot;C&quot;,&quot;G&quot;)</code></pre>
<p class="mb-3">They can also contain multi-letter <em>strings</em>:</p>
<pre class="sourcecode"><code>my.oligos &lt;- c(&quot;ATCGC&quot;,&quot;TTTCGC&quot;,&quot;CCCGCG&quot;,&quot;GGGCGC&quot;)</code></pre>
<h2 class="text-xl mt-5 mb-5" data-number="2.3" id="tables"><span class="header-section-number">2.3</span> Tables</h2>
<p class="mb-3">If we made a vector variable ‚Äúnucleotides‚Äù containing the of a DNA molecule, we can use the <code>table()</code> function to produce a <strong>table variable</strong> that contains the number of bases with each possible nucleotides:</p>
<pre class="sourcecode"><code>bases &lt;- c(&quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;G&quot;)</code></pre>
<p class="mb-3">Now make the table</p>
<pre class="sourcecode"><code>&gt; table(bases)
bases
A C G T 
3 2 2 2</code></pre>
<h2 class="text-xl mt-5 mb-5" data-number="2.4" id="arguments"><span class="header-section-number">2.4</span> Arguments</h2>
<p class="mb-3">Functions in R usually require <em>arguments</em>, which are input variables (i.e.. objects) that are <em>passed</em> to them, which they then carry out some operation on. For example, the <code>log10()</code> function is passed a number, and it then calculates the log to the base 10 of that number:</p>
<pre class="sourcecode"><code>&gt; log10(100)
[1] 2</code></pre>
<p class="mb-3">There‚Äôs a more generic function, <code>log()</code>, where we pass it not only a number to take the log of, but also the specific <code>base</code> of the logarithm. To take the log base 10 with the <code>log()</code> function we do this</p>
<pre class="sourcecode"><code>&gt; log(100, base = 10)
[1] 2</code></pre>
<p class="mb-3">We can also take logs with other bases, such as 2:</p>
<pre class="sourcecode"><code>&gt; log(100, base = 2)
[1] 6.643856</code></pre>
<p class="mb-3">We can perform computations with R using objects such as scalars and vectors. For example, to calculate the average of the values in the vector <code>myvector</code> (i.e.. the average of 8, 6, 9, 10 and 5), we can use the <code>mean()</code> function:</p>
<pre class="sourcecode"><code>&gt; mean(myvector) # note: no &quot; &quot;
[1] 7.6</code></pre>
<p class="mb-3">We have been using built-in R functions such as <code>mean()</code>, <code>length()</code>, <code>print()</code>, <code>plot()</code>, etc.</p>
<h2 class="text-xl mt-5 mb-5" data-number="2.5" id="writing-your-own-functions"><span class="header-section-number">2.5</span> Writing your own functions</h2>
<p class="mb-3">We can also create our own functions in R to do calculations that you want to carry out very often on different input data sets. For example, we can create a function to calculate the value of 20 plus square of some input number, in the code panel:</p>
<pre class="sourcecode"><code>myfunction &lt;- function(x) {
    y &lt;- x * x
    return(20 + y)
}</code></pre>
<p class="mb-3">This function will calculate the square of a number <code>x</code>, and then add 20 to that value. The <code>return()</code> statement returns the calculated value. Once you have typed in this function, the function is then available for use. For example, we can use the function for different input numbers (e.g.. 10, 25):</p>
<pre class="sourcecode"><code>&gt; myfunction(10)
[1] 120</code></pre>
<h2 class="text-xl mt-5 mb-5" data-number="2.6" id="conditionals"><span class="header-section-number">2.6</span> Conditionals</h2>
<p class="mb-3">It is often useful to execute different operations depending on a condition. Let‚Äôs say we want to say if a number if big (larger than 10) or small (otherwise):</p>
<pre class="sourcecode"><code>x &lt;- 5
if (x &gt; 10) {
  print(&quot;Big!&quot;)
} else {
  print(&quot;Small!&quot;)
}</code></pre>
<p class="mb-3">By changing the value of <code>x</code>, you can see either the first or the second piece of code being executed.</p>
<h2 class="text-xl mt-5 mb-5" data-number="2.7" id="loops"><span class="header-section-number">2.7</span> Loops</h2>
<p class="mb-3">Finally, sometimes we want to execute an operation many times. The way to do it is using a <code>for</code> loop. For example, to print the numbers from 1 to 10 we can do:</p>
<pre class="sourcecode"><code>for (x in 1:10) {  
¬† print(x)  
}</code></pre>
<p class="mb-3">Or, to print every element of an array:</p>
<pre class="sourcecode"><code>dice &lt;- c(1, 2, 3, 4, 5, 6)

for (x in dice) {  
¬† print(x)  
}</code></pre>

  <ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul>


</article>
  <script type="text/javascript">
  function googleTranslateElementInit() {
    new google.translate.TranslateElement({
      pageLanguage: 'en',
      autoDisplay: false,
      includedLanguages: 'en,ru,uz',
    }, 'google_translate_element');
  }
  </script>

  <script type="text/javascript" src="https://translate.google.com/translate_a/element.js?cb=googleTranslateElementInit"></script>
</div>

</body>
</html>
