<!DOCTYPE html>
<html lang="en">
<head profile="http://www.w3.org/2005/10/profile">
    <meta charset="UTF-8">
    <meta name="description" content="">
    <meta name="author" content="">
    <meta name=viewport content="width=device-width, initial-scale=1">
    <title>1.2 Comparing Gene Sequences
</title>
    <link rel="stylesheet" href="/css/style.css">
    <link rel="stylesheet" href="/css/syntax.css">
    <link href="https://cdn.jsdelivr.net/npm/daisyui@3.1.0/dist/full.css" rel="stylesheet" type="text/css" />
    <script src="https://cdn.tailwindcss.com"></script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.css" integrity="sha384-GvrOXuhMATgEsSwCs4smul74iXGOixntILdUW9XmUC6+HX0sLNAK3q71HotJqlAn" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.js" integrity="sha384-cpW21h6RZv/phavutF+AuVYrr+dA8xD9zs6FwLpaCct6O9ctzYFfFr4dgmgccOTx" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/contrib/auto-render.min.js" integrity="sha384-+VBxd3r6XgURycqtZ117nYw44OOcIax56Z4dCRWbxyPt0Koah1uHoK0o4+/RRE05" crossorigin="anonymous"
        onload="renderMathInElement(document.body);"></script>

</head>
<body>
    <header>
        <div class="navbar bg-base-100">
          <div class="navbar-start">
            <nav>
            <ol class="inline-flex items-center space-x-1 md:space-x-3">
        <li class="inline-flex items-center">
          <a href="/" class="inline-flex items-center text-l font-medium text-gray-700 hover:text-blue-600 dark:text-gray-400 dark:hover:text-white">
            <svg aria-hidden="true" class="w-4 h-4 mr-2" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"></path></svg>
            Juan Raphael Diaz Simões
          </a>
        </li>
    
        <li>
      <div class="flex items-center">
        <svg aria-hidden="true" class="w-6 h-6 text-gray-400" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" clip-rule="evenodd"></path></svg>
        <a href="/courses" class="ml-1 text-l font-medium text-gray-700 hover:text-blue-600 md:ml-2 dark:text-gray-400 dark:hover:text-white">Courses</a>
      </div>
    </li>

          </div>
          <div class="navbar-end">
            <ul class="menu menu-horizontal px-1">
              <li><a href="/posts">Posts</a></li>
              <li><a href="/courses">Courses</a></li>
              <li><a href="/cv.html">CV</a></li>
              <li><a href="/projects.html">Projects</a></li>
            </ul>
          </div>
        </nav>
    </header>
    <div class="flex justify-center bg-base-100">
<article class="prose max-w-prose mb-5">
  <div class="book-navigation">

    <div class="book-navigation-parent drawer">
      <input id="my-drawer" type="checkbox" class="drawer-toggle" />
      <div class="drawer-content">
        <label for="my-drawer">
          <div class="space-y-1 inline-block">
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
            <div class="w-4 h-0.5 bg-gray-600"></div>
          </div>
          <b>Index</b>
        </label>
      </div>
      <div class="drawer-side">
        <label for="my-drawer" class="drawer-overlay"></label>
           <div class="menu menu-vertical p-4 bg-base-200">
             <h1 class="text-xl mb-5 w-100" style="margin-left: 0.75em;">
                  Introduction to Bioinformatics

             </h1>
             <div style="margin-left:-1.5em;">
               <ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/index.html">1 Lectures
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/03-lecture-01/index.html">1.1 Bioinformatics Basics
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/07-lecture-02/index.html">1.2 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/10-lecture-03/index.html">1.3 Comparing Gene Sequences 2
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/13-lecture-04/index.html">1.4 Gene Finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/01-lectures/16-lecture-05/index.html">1.5 Locating genes and its functions: techniques, problems and difficulties
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/index.html">2 Practical Lessons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/index.html">2.1 Basic R and generic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/01-seminar-01/index.html">2.1.1 Basic R
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/02-seminar-02/index.html">2.1.2 Files
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/04-seminar-03/index.html">2.1.3 Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/05-seminar-04/index.html">2.1.4 Comparing Gene Sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/06-control-01/index.html">2.1.5 Loading and processing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/01-basic-r-and-fasta/20-seminar-12/index.html">2.1.6 Programming Exercises
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/index.html">2.2 Basic analysis of genetic data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/08-seminar-05/index.html">2.2.1 GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/09-seminar-06/index.html">2.2.2 Subseries GC Content
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/11-seminar-07/index.html">2.2.3 Creating dot plots
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/02-gc-content-dot-plot/14-control-02/index.html">2.2.4 Loading and comparing data
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/index.html">2.3 Codons and gene finding
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/15-seminar-09/index.html">2.3.1 Translating bases to codons
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/17-seminar-10/index.html">2.3.2 Codon frequencies and finding genes
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/18-control-03/index.html">2.3.3 Theoretical Control
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/03-codons-and-gene-finding/19-seminar-11/index.html">2.3.4 Aminoacid sequences and base sequences
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/index.html">2.4 Summary and exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/21-seminar-13/index.html">2.4.1 Theoretical Questions
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li><li><a class="link link-hover text-zinc-600" href="/courses/bioinformatics/02-practical-lessons/04-summary-and-exam/22-exam-01/index.html">2.4.2 Final exam
</a><ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul></li></ul></li></ul></li></ul>
             </div>
           </div>
      </div>
    </div>

    <div class="book-navigation-parent">
      <div class="arrow arrow-up"></div>
      <a href="/courses/bioinformatics/01-lectures/index.html">1 Lectures
</a>
      <div class="arrow arrow-up"></div>
    </div>

    <div class="book-navigation-section">

       <div class="book-navigation-previous">
         <div class="arrow arrow-left"></div>
         <a href="/courses/bioinformatics/01-lectures/03-lecture-01/index.html">1.1 Bioinformatics Basics
</a>
       </div>

       <div class="book-navigation-next">
         <a href="/courses/bioinformatics/01-lectures/10-lecture-03/index.html">1.3 Comparing Gene Sequences 2
</a>
         <div class="arrow arrow-right"></div>
       </div>
    </div>
  </div>

    <h1 class="text-3xl mb-5">
    1.2 Comparing Gene Sequences

</h1>

<h1 class="text-2xl mt-5 mb-5" data-number="1" id="context"><span class="header-section-number">1</span> Context</h1>
<h2 class="text-xl mt-5 mb-5" data-number="1.1" id="bases"><span class="header-section-number">1.1</span> Bases</h2>
<p class="mb-3">DNA is composed of four bases adenine (<strong>A</strong>), guanine (<strong>G</strong>), cytosine (<strong>C</strong>) and thymine (<strong>T</strong>).</p>
<p class="mb-3">Bases match in the DNA in two pairs: <strong>G-C</strong> and <strong>A-T</strong>.</p>
<p class="mb-3"><img class="mx-auto" src="data/2598c30f7e7dbc62f1c16dde34714bd5.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="1.2" id="bases-sequences"><span class="header-section-number">1.2</span> Bases sequences</h2>
<p class="mb-3">DNA is a sequence of such bases. The bases of DNA determines the RNA it can encode, and therefore the proteins in can enduce production of.</p>
<p class="mb-3"><img class="mx-auto" src="data/1b1f5ce9f25913897c5ae987912331cd.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="1.3" id="question"><span class="header-section-number">1.3</span> Question</h2>
<p class="mb-3">Given two sequences, how can we determine in which aspects they are <strong>different</strong> or <strong>similar</strong>?</p>
<p class="mb-3">Differences are simple to find, but not all of them determine <strong>functional</strong> differences.</p>
<p class="mb-3">In the same way, not all similarities we can find have <strong>biological</strong> meaning.</p>
<p class="mb-3">I will show some techniques that are used currently.</p>
<h1 class="text-2xl mt-5 mb-5" data-number="2" id="gc-content"><span class="header-section-number">2</span> GC Content</h1>
<h2 class="text-xl mt-5 mb-5" data-number="2.1" id="bonds"><span class="header-section-number">2.1</span> Bonds</h2>
<p class="mb-3">The <strong>A-T</strong> bond has two hydrogen bonds, while the <strong>G-C</strong> one has three. Often we write these bonds as <strong>A=T</strong> and <strong>G≡C</strong> to emphasize the difference.</p>
<p class="mb-3">The <strong>G-C</strong> bond is more thermostable and allows more stable stackings on the DNA strand.</p>
<p class="mb-3">DNA with low GC-content is less stable than DNA with high GC-content. Especially at high temperatures.</p>
<p class="mb-3"><img class="mx-auto" src="data/a93c9a90daeff4ee5073a323986bde16.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="2.2" id="calculation"><span class="header-section-number">2.2</span> Calculation</h2>
<p class="mb-3">The GC content is calculated as:</p>
<p class="mb-3"><img class="mx-auto" src="data/164df7f5f092e4529a73b266a16b6571.png" /></p>
<p class="mb-3">It can be measured by observing the melting temperature of the DNA double helix using spectrophotometry. The absorbance of DNA at a wavelength of 260 nm increases when the double-stranded DNA molecule separates into two single strands.</p>
<h2 class="text-xl mt-5 mb-5" data-number="2.3" id="calculating-from-sequence"><span class="header-section-number">2.3</span> Calculating from sequence</h2>
<p class="mb-3">GC content can be calculated directly from the gene sequence. It is enough to sum the amounts of G and C and divide by the total amount.</p>
<div class="sourceCode mb-3 notranslate" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a class="link link-hover text-zinc-600" href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> dengue <span class="ot">&lt;-</span> <span class="fu">read.fasta</span>(<span class="st">&quot;dengue.fasta&quot;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb1-2"><a class="link link-hover text-zinc-600" href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> c1 <span class="ot">&lt;-</span> <span class="fu">count</span>(dengue,<span class="dv">1</span>)</span>
<span id="cb1-3"><a class="link link-hover text-zinc-600" href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> c1</span>
<span id="cb1-4"><a class="link link-hover text-zinc-600" href="#cb1-4" aria-hidden="true" tabindex="-1"></a>   a    c    g    t </span>
<span id="cb1-5"><a class="link link-hover text-zinc-600" href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="dv">3426</span> <span class="dv">2240</span> <span class="dv">2770</span> <span class="dv">2299</span> </span>
<span id="cb1-6"><a class="link link-hover text-zinc-600" href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> (c1[<span class="st">&quot;c&quot;</span>] <span class="sc">+</span> c1[<span class="st">&quot;g&quot;</span>]) <span class="sc">/</span> <span class="fu">sum</span>(c1)</span>
<span id="cb1-7"><a class="link link-hover text-zinc-600" href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fl">0.4666977</span> </span></code></pre></div>
<p class="mb-3">Threfore, the dengue sequence has around 47% of GC content.</p>
<h2 class="text-xl mt-5 mb-5" data-number="2.4" id="using-seqinr"><span class="header-section-number">2.4</span> Using <code>seqinr</code></h2>
<p class="mb-3">The library <code>seqinr</code> has a function that automatically calculates the GC content for any sequence.</p>
<div class="sourceCode mb-3 notranslate" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a class="link link-hover text-zinc-600" href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">GC</span>(dengue)</span>
<span id="cb2-2"><a class="link link-hover text-zinc-600" href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fl">0.4666977</span></span></code></pre></div>
<h2 class="text-xl mt-5 mb-5" data-number="2.5" id="selecting-windows"><span class="header-section-number">2.5</span> Selecting windows</h2>
<p class="mb-3">Different parts of the DNA strand behave differently. It is useful to be able to select subparts of it. For example, to select the first 10 nucleotides:</p>
<div class="sourceCode mb-3 notranslate" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a class="link link-hover text-zinc-600" href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> dengue[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb3-2"><a class="link link-hover text-zinc-600" href="#cb3-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">&quot;a&quot;</span> <span class="st">&quot;g&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;g&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;a&quot;</span> <span class="st">&quot;g&quot;</span> <span class="st">&quot;t&quot;</span></span></code></pre></div>
<p class="mb-3">To select <code>n</code> nucleotides, starting from the position <code>s</code>, the operation to be used is <code>dengue[s,(s+n-1)]</code>.</p>
<h2 class="text-xl mt-5 mb-5" data-number="2.6" id="gc-content-in-subsequences"><span class="header-section-number">2.6</span> GC content in subsequences</h2>
<p class="mb-3">The GC content on a sequence is not always the same depending on the place of the sequence.</p>
<p class="mb-3">We want to see the GC content on different nucleotide windows. For example, from the 1st to the 1000th, from the 1001st to the 2000th, etc… For this, we define a function:</p>
<div class="sourceCode mb-3 notranslate" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a class="link link-hover text-zinc-600" href="#cb4-1" aria-hidden="true" tabindex="-1"></a>windowGC <span class="ot">&lt;-</span> <span class="cf">function</span>(data, window) {</span>
<span id="cb4-2"><a class="link link-hover text-zinc-600" href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  starts <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="fu">length</span>(data)<span class="sc">-</span>window,<span class="at">by=</span>window)</span>
<span id="cb4-3"><a class="link link-hover text-zinc-600" href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-4"><a class="link link-hover text-zinc-600" href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(starts, <span class="cf">function</span>(start) {</span>
<span id="cb4-5"><a class="link link-hover text-zinc-600" href="#cb4-5" aria-hidden="true" tabindex="-1"></a>    range <span class="ot">&lt;-</span> start<span class="sc">:</span>(start<span class="sc">+</span>window<span class="dv">-1</span>)</span>
<span id="cb4-6"><a class="link link-hover text-zinc-600" href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">GC</span>(data[range])</span>
<span id="cb4-7"><a class="link link-hover text-zinc-600" href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb4-8"><a class="link link-hover text-zinc-600" href="#cb4-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<h2 class="text-xl mt-5 mb-5" data-number="2.7" id="gc-content-in-subsequences-1"><span class="header-section-number">2.7</span> GC content in subsequences</h2>
<p class="mb-3">To calculate for windows of size 1000:</p>
<div class="sourceCode mb-3 notranslate" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a class="link link-hover text-zinc-600" href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">windowGC</span>(dengue,<span class="dv">1000</span>)</span>
<span id="cb5-2"><a class="link link-hover text-zinc-600" href="#cb5-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="fl">0.477</span> <span class="fl">0.453</span> <span class="fl">0.464</span> <span class="fl">0.441</span> <span class="fl">0.471</span> <span class="fl">0.470</span> <span class="fl">0.488</span> <span class="fl">0.470</span> <span class="fl">0.438</span> <span class="fl">0.471</span></span>
<span id="cb5-3"><a class="link link-hover text-zinc-600" href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">barplot</span>(<span class="fu">windowGC</span>(dengue,<span class="dv">1000</span>))</span></code></pre></div>
<p class="mb-3"><img class="mx-auto" src="data/168364e27ee6e0dc681bf322607d446d.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="2.8" id="different-window-sizes"><span class="header-section-number">2.8</span> Different window sizes</h2>
<p class="mb-3">The function above can be used to compare different window sizes. By decreasing the window size, we can see the GC content with more resolution:</p>
<p class="mb-3"><img class="mx-auto" src="data/c1e1d16948eee263d8373a848e960783.png" /></p>
<p class="mb-3">Looking at this graph, what part of the DNA would break first when heating it?</p>
<h1 class="text-2xl mt-5 mb-5" data-number="3" id="dot-plots"><span class="header-section-number">3</span> Dot plots</h1>
<h2 class="text-xl mt-5 mb-5" data-number="3.1" id="what-is-a-dot-plot"><span class="header-section-number">3.1</span> What is a dot plot</h2>
<p class="mb-3">A dot plot is a graphical method for comparing two biological sequences and identifying regions of close similarity after sequence alignment. It is one of the oldest techniquest for comparing base sequences.</p>
<p class="mb-3">The technique is the following. Build a graph with two axis, with one gene sequence in each axis. In the position <code>(i,j)</code> put a black dot if the base in the position <code>i</code> of the first sequence is the same as the base in the position <code>j</code> in the second sequence. Othewise, put a white dot.</p>
<p class="mb-3"><img class="mx-auto" src="data/983b437ea763b2228443d2be2ad544dd.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="3.2" id="how-to-make-a-dot-plot"><span class="header-section-number">3.2</span> How to make a dot plot</h2>
<p class="mb-3">In R, is is possible to make a dot plot using the <code>seqinr</code> library. To do a dot plot for a small part of the dengue gene:</p>
<div class="sourceCode mb-3 notranslate" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a class="link link-hover text-zinc-600" href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> shortDengue <span class="ot">&lt;-</span> dengue[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb6-2"><a class="link link-hover text-zinc-600" href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">dotPlot</span>(shortDengue,shortDengue)</span></code></pre></div>
<p class="mb-3">Why a small section? Dot plots are expensive to calculate. For 100 bases, one needs 10,000 operations. For 10,000 bases, one needs 100,000,000 operations. It grows very fast, so it can take a long time to calculate.</p>
<p class="mb-3"><img class="mx-auto" src="data/00ce6669265d5e06ec115509582cc3cc.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="3.3" id="comparing-a-sequence-with-itself"><span class="header-section-number">3.3</span> Comparing a sequence with itself</h2>
<p class="mb-3">Diagonal black lines show parts of both sequences that are the same.</p>
<p class="mb-3">In this case, we are comparing a sequence to itself. Therefore, the main diagonal of the square is completely black. However, other diagonals can be seen in the graph. This means some chain is also happening elsewhere on the sequence.</p>
<p class="mb-3">What about the black block we see around the indexes 65 to 80? We can see it is a sequence of <code>T</code>’s.</p>
<div class="sourceCode mb-3 notranslate" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a class="link link-hover text-zinc-600" href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> shortDengue[<span class="dv">65</span><span class="sc">:</span><span class="dv">80</span>]</span>
<span id="cb7-2"><a class="link link-hover text-zinc-600" href="#cb7-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">&quot;a&quot;</span> <span class="st">&quot;a&quot;</span> <span class="st">&quot;c&quot;</span> <span class="st">&quot;a&quot;</span> <span class="st">&quot;g&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;a&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;t&quot;</span> <span class="st">&quot;a&quot;</span> <span class="st">&quot;g&quot;</span></span></code></pre></div>
<p class="mb-3"><img class="mx-auto" src="data/00ce6669265d5e06ec115509582cc3cc.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="3.4" id="comparing-the-full-sequence"><span class="header-section-number">3.4</span> Comparing the full sequence</h2>
<p class="mb-3">When comparing the full dengue sequence with itself, the result is the following. It took around 20 minutes to calculate in my computer, so it is not practical to do during classes.</p>
<p class="mb-3">It also shows another limitation of dot plots – when they are very large, it is difficult to interpret the result, since lines are difficult to see.</p>
<p class="mb-3"><img class="mx-auto" src="data/020eeab53cadec564debb6dbc2bdf69f.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="3.5" id="comparing-two-sequences"><span class="header-section-number">3.5</span> Comparing two sequences</h2>
<p class="mb-3">Besides comparing a sequence with itself, we can compare two different sequences:</p>
<div class="sourceCode mb-3 notranslate" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a class="link link-hover text-zinc-600" href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> shortZika <span class="ot">&lt;-</span> zika[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb8-2"><a class="link link-hover text-zinc-600" href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">dotPlot</span>(shortDengue,shortZika)</span></code></pre></div>
<p class="mb-3"><img class="mx-auto" src="data/536788fb6455178c3f33cebd52c681d2.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="3.6" id="comparing-a-larger-sequence"><span class="header-section-number">3.6</span> Comparing a larger sequence</h2>
<p class="mb-3">A larger dot plot for comparing dengue and zika, using the first 1000 bases, is the following. As before, it took a long time to calculate, and it is not very easy to interpret.</p>
<p class="mb-3"><img class="mx-auto" src="data/be80e960c130a54d7da2860158474dbb.png" /></p>
<h2 class="text-xl mt-5 mb-5" data-number="3.7" id="next"><span class="header-section-number">3.7</span> Next</h2>
<p class="mb-3">This is a preview of what we will do in the next lectures. We will calculate GC contents and dot plots using R and the gene sequences we already have.</p>
<h2 class="text-xl mt-5 mb-5" data-number="3.8" id="questions"><span class="header-section-number">3.8</span> Questions?</h2>

  <ul class="list-disc list-outside mt-1 mb-3 pl-5"></ul>


</article>
</div>

</body>
</html>
